liquibase-core/src/main/java/liquibase/diff/output/changelog/DiffToChangeLog.java

```diff
@@ -91,23 +92,38 @@ public void print(PrintStream out) throws ParserConfigurationException, IOExcept
 
     public void print(String changeLogFile, ChangeLogSerializer changeLogSerializer) throws ParserConfigurationException, IOException, DatabaseException {
         this.changeSetPath = changeLogFile;
-        File file = new File(changeLogFile);
+        final PathHandlerFactory pathHandlerFactory = Scope.getCurrentScope().getSingleton(PathHandlerFactory.class);
+        Resource resource = pathHandlerFactory.getResource(changeLogFile, true);
+        boolean fileExists;
+        // todo, an exists method would be useful here, instead of this hack
+        try (InputStream is = resource.openInputStream()) {
+            fileExists = is != null;
+        } catch (IOException | NullPointerException e) {
+            fileExists = false;
+            Scope.getCurrentScope().getLog(getClass()).warning("Unable to open input stream to file " + changeLogFile, e);
+        }
```

Comment:
Group agrees that this should be replaced with an `exists(String path)` method in the PathHandler interface.